File: arosaje-frontend/src/Domain/TicketComment.js
class TicketComment{
    id: number;
    ticket: Ticket;
    user: User;
    comment: string;
    createdAt: Date;

    constructor(id: number, ticket: Ticket, user: User, comment: string, createdAt: Date) {
        this._id = id;
        this._ticket = ticket;
        this._user = user;
        this._comment = comment;
        this._createdAt = createdAt;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get ticket(): Ticket {
        return this._ticket;
    }

    set ticket(value: Ticket) {
        this._ticket = value;
    }

    get user(): User {
        return this._user;
    }

    set user(value: User) {
        this._user = value;
    }

    get comment(): string {
        return this._comment;
    }

    set comment(value: string) {
        this._comment = value;
    }

    get createdAt(): Date {
        return this._createdAt;
    }

    set createdAt(value: Date) {
        this._createdAt = value;
    }
}

File: arosaje-backend/src/main/java/fr/epsi/mspr/arosaje/entity/TicketComment.java
package fr.epsi.mspr.arosaje.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "ticket_comments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TicketComment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "ticket_id", referencedColumnName = "id")
    private Ticket ticket;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "ticketComment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Photo> photos;
}
